hint.cabal

```diff
@@ -40,14 +40,13 @@ test-suite unit-tests
                   HUnit,
                   directory,
                   filepath,
-                  extensible-exceptions,
```

Comment:
this is an unrelated improvement: the documentation for `extensible-exceptions` says that from ghc-6.9 onwards, it simply re-exports `base`'s exception module. Since hint only supports ghc-8.6 and up, we definitely don't need that old compatibility layer anymore.